addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -38,6 +38,26 @@ type ManagedClusterAddOnSpec struct {
 	// +kubebuilder:validation:MaxLength=63
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	InstallNamespace string `json:"installNamespace,omitempty"`
+
+	// healthCheck defines how to check the healthiness status of the current addon.
+	// If not set, the lease mode will be used
+	// +optional
+	HealthCheck HealthCheck `json:"healthCheck,omitempty"`
```

Comment:
I think this should be in the status field. User should not set this.

Reply:
move to status field